portfolio/src/main/webapp/script.js

```diff
@@ -16,7 +16,13 @@
  * Adds comments from /data
  */
 async function getComments() {
-    const commentsData = await fetch("/data");
-    const comments = await commentsData.text();
-    document.getElementById("comments-container").innerHTML = comments;
+  const commentsData = await fetch('/data');
+  const comments = await commentsData.json();
+  const ul = document.createElement('ul');
+  for (let i = 0; i < comments.length; i++) {
```

Comment:
Prefer ES6 style
for (let [comment, index] of comments) {...}

The square brackets are called destructuring.
https://exploringjs.com/es6/ch_for-of.html